src/include/simeng/RegisterValue.hh

```diff
@@ -90,11 +96,16 @@ class RegisterValue {
    * the specified datatype. */
   template <class T>
   const T* getAsVector() const {
-    static_assert(alignof(T) <= 8 && "Alignment over 8 bytes not guranteed");
+    static_assert(alignof(T) <= 8 && "Alignment over 8 bytes not guaranteed");
     assert(bytes > 0 && "Attempted to access an uninitialised RegisterValue");
-    assert(sizeof(T) <= bytes &&
-           "Attempted to access a RegisterValue as a datatype larger than the "
-           "data held");
+    assert((sizeof(T) <= bytes || (bytes == 4 && sizeof(T) == 8)) && "Attempted"
```

Comment:
Why has the second condition been added in the assert and why is it specific to only `bytes == 4 && sizeof(T) == 8`? Surely the first expression in the assert i.e. `sizeof(T) <= bytes` incorporates the second expression.